File: REST/core/e1a.pl
#!/bin/env perl

# Use the /lookup endpoint to find IRAK4

use strict;
use warnings;

use HTTP::Tiny;
use JSON;
use Data::Dumper;
$Data::Dumper::Terse = 1;
$Data::Dumper::Indent = 1;

my $server = "http://rest.ensembl.org";
my $lookup_endpoint = "/lookup/symbol/human/%s";

my $http = HTTP::Tiny->new();

# Fetch an endpoint from the serverm allow overriding of the default content type
sub fetch_endpoint {
    my ($request, $content_type) = @_;
    $content_type ||= 'application/json';

    my $response = $http->get($request,
                              { headers => { 'Content-type' => $content_type } } );

    die "Failed!\n" unless $response->{success};

    if($content_type eq 'application/json') {
	return decode_json($response->{content});
    } else {
	return $response->{content};
    }
}

my $url = $server . sprintf($lookup_endpoint, 'IRAK4');
my $gene = fetch_endpoint($url);

print Dumper $gene;
print "\n";

File: REST/core/e1a.py
import requests, sys, json
import pprint

'''
Use the /lookup endpoint to find IRAK4
'''

server = "http://rest.ensembl.org"
lookup_endpoint = "/lookup/symbol/human/{}"

def fetch_json(server, request):
    """
    Fetch an endpoint from the server and return the decoded JSON results.
    """
    r = requests.get(server+request, headers={ "Content-Type" : "application/json"})

    if not r.ok:
        r.raise_for_status()
        sys.exit()

    return r.json()

if __name__ == "__main__":

    # Fetch the record for the gene IRAK4
    gene = fetch_json(server, lookup_endpoint.format('IRAK4'))

print(json.dumps(gene, indent=4, sort_keys=True))
    #pprint.pprint(gene)
